src/core/execute.c

```diff
@@ -708,22 +714,48 @@ static int restore_confirm_stdio(int *saved_stdin, int *saved_stdout) {
         return r;
 }
 
-static int ask_for_confirmation(const char *vc, char *response, char **argv) {
+typedef enum ConfirmAnswer {
+        CONFIRM_PRETEND_FAILURE = -1,
+        CONFIRM_PRETEND_SUCCESS =  0,
+        CONFIRM_EXECUTE = 1,
+} ConfirmAnswer;
+
+static ConfirmAnswer ask_for_confirmation(const char *vc, const char *cmdline) {
```

Comment:
Hmm, can't say I like the fact that the prototype of this function suggests we return an enum here, but actually you store the result before returning it in an "int". Stick to one type: either use the right enum type for all internal variables that use the enum values. Or make this an anonymous enum, i.e. use "int" as type for this everywhere, but define possible values for this via an enum that defines no type...